harambe/parser/type_currency.py

```diff
@@ -10,14 +10,31 @@ def __new__(cls) -> Any:
 
     @staticmethod
     def validate_currency(value: str) -> float:
-        if isinstance(value, float):
-            return value
+        if isinstance(value, (float, int)):
+            return float(value)
 
         value = str(value).strip()
 
-        value = re.sub(r"[^\d.-]", "", value)
+        if ":" in value:  # cases like 12:00
+            value = re.sub(r"(?<=\d):(?=\d)", ".", value)
+        cleaned_value = re.sub(r"[^\d.,-]", "", value)
+        cleaned_value = re.sub(r"^0+(?!$)", "", cleaned_value)
+        if "." in cleaned_value:
+            decimal_parts = cleaned_value.split(".")
+            if len(decimal_parts[-1]) == 3:  # thousands separators
+                cleaned_value = cleaned_value.replace(".", "")
+        if cleaned_value.startswith("."):  # fraction
+            cleaned_value = "0" + cleaned_value
+            return float(cleaned_value)
+        if "," in cleaned_value and "." in cleaned_value:
+            if cleaned_value.index(",") < cleaned_value.index("."):
+                cleaned_value = cleaned_value.replace(",", "")
+            else:
+                cleaned_value = cleaned_value.replace(".", "").replace(",", ".")
+        elif "," in cleaned_value and "." not in cleaned_value:
+            cleaned_value = cleaned_value.replace(",", "")
```

Comment:
This is a big blob of code. lets split it up for readability

```suggestion
        if ":" in value:  # cases like 12:00
            value = re.sub(r"(?<=\d):(?=\d)", ".", value)
            
        cleaned_value = re.sub(r"[^\d.,-]", "", value)
        cleaned_value = re.sub(r"^0+(?!$)", "", cleaned_value)
        
        if "." in cleaned_value:
            decimal_parts = cleaned_value.split(".")
            if len(decimal_parts[-1]) == 3:  # thousands separators
                cleaned_value = cleaned_value.replace(".", "")
                
        if cleaned_value.startswith("."):  # fraction
            cleaned_value = "0" + cleaned_value
            return float(cleaned_value)
            
        if "," in cleaned_value and "." in cleaned_value:
            if cleaned_value.index(",") < cleaned_value.index("."):
                cleaned_value = cleaned_value.replace(",", "")
            else:
                cleaned_value = cleaned_value.replace(".", "").replace(",", ".")
        elif "," in cleaned_value and "." not in cleaned_value:
            cleaned_value = cleaned_value.replace(",", "")
```

Reply:
yeah you are right, it is getting cloudy